site/src/sphinx/server-annotated-service.rst

```diff
@@ -0,0 +1,627 @@
+
+    class MyDecorator implements DecoratingServiceFunction<HttpRequest, HttpResponse> {
+        @Override
+        public HttpResponse serve(Service<HttpRequest, HttpResponse> delegate,
+                                  ServiceRequestContext ctx,
+                                  HttpRequest req) throws Exception {
+            ... Do something ...
+            return delegate.serve(ctx, req);
+        }
+    }
+
+Then, annotate your class or method with `@Decorator`_ annotation. In the following example, ``MyDecorator``
+will handle a request first, then ``AnotherDecorator`` will handle the request next, and finally ``hello()``
+method will handle the request.
+
+.. code-block:: java
+
+    @Decorator(MyDecorator.class)
+    class MyAnnotatedServiceClass {
+
+        @Decorator(AnotherDecorator.class)
+        @Get("/hello")
+        public HttpResponse hello() {
+            ...
+        }
+    }
+
+.. note::
+
+    The applying order of the decorators is slightly different to that of the converters and exception handlers.
+
+
+.. _media_type_nego:
+
+Media type negotiation
+----------------------
+
+*Armeria* provides `@ProduceType`_ and `@ConsumeType`_ annotations to support media type negotiation. It is not
+necessary if you have only one service method for a path and an HTTP method. Assume that you have multiple
+service methods for the same path and the same HTTP method as follows.
+
+.. code-block:: java
+
+    class MyAnnotatedServiceClass {
+
+        @Get("/hello")
+        public HttpResponse hello1() {
+            // Return a text document to the client.
+        }
+
+        @Get("/hello")
+        public HttpResponse hello2() {
+            // Return a JSON object to the client.
+        }
+    }
+
+If no media type is specified on the method, the first method is usually mapped to a request for the path
+``/hello``. So you would always receive a text document from a server. What should you do if you want to
+get a JSON object from the path ``/hello``? You can just specify the type of the content what your method
+produces as follows and add ``Accept`` header to your request.
+
+.. code-block:: java
+
+    class MyAnnotatedServiceClass {
+
+        @Get("/hello")
+        @ProduceType("text/plain")
+        public HttpResponse helloText() {
+            // Return a text document to the client.
+        }
+
+        @Get("/hello")
+        @ProduceType("application/json")
+        public HttpResponse helloJson() {
+            // Return a JSON object to the client.
+        }
+    }
+
+A request like the following would be get a text document::
+
+    GET /hello HTTP/1.1
+    Accept: text/plain
+
+A request like the following would be get a JSON object::
+
+    GET /hello HTTP/1.1
+    Accept: application/json
+
+.. note::
+
+    Note that ``Content-Type`` header of a response is not automatically set. You are able to get the
+    negotiated `@ProduceType`_ from ``ServiceRequestContext.negotiatedProduceType()`` method.
```

Comment:
That being said, we never showed an example that injects `ServiceRequestContext`. Could you add a section about the types of automatically injected params? e.g. `ServiceRequestContext`, `HttpRequest`, `AggregatedHttpMessage`